models/stdp_dopamine_synapse.cpp

```diff
@@ -31,6 +31,8 @@
 // Includes from sli:
 #include "dictdatum.h"
 
+#define ClassToString( cls ) #cls
```

Comment:
How does this macro work?

Reply:
I think it's just a matter of performance, but now that Charl mentioned the problem with these changes, I don't think this change makes sense anymore, even if it would change performance at all. Not that this part of the code would be performance-critical/-relevant at all.

Reply:
The macro simply takes the input and interprets it as the actual characters that were given to that macro. So all it does is change `volume_transmitter` to `"volume_transmitter"`.

Reply:
If we can apply certain checks if the underlying `Node` object can be converted to `volume_transmitter`, we can avoid using the `dynamic_cast` and therefore avoid the runtime overhead used for the conversion.

Reply:
You would have to perform some profiling, though. All the overhead required for the manual type-checking might actually outweigh the overhead of the `dynamic_cast`, but calling the RTTI might actually not consume that much more time. The branching will be what costs us a lot of cycles as well, and the if will still persist with your solution. So in total there might not even be an advantage of doing manual type-checking here.